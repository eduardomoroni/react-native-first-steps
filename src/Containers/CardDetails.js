/* @flow */

import React, { Component } from 'react'
import {
  View,
  StyleSheet,
  Image
} from 'react-native'
import {
  Metrics
} from '../Styles/Themes'

type CardDetailsProps = {
  card: any
}

export default class CardDetails extends Component {
  props: CardDetailsProps

  render () {
    const {
      multiverseid
    } = this.props.card
    const IMG_URL = `http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=${multiverseid}&type=card`

    return (
      <View style={styles.container}>
        <Image style={styles.image} source={{uri: IMG_URL}} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: Metrics.navBarHeight
  },
  image: {
    flex: 1
  }
})
