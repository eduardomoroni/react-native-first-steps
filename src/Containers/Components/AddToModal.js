import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import { cardType } from '../../Types/CardType'
import { Modal } from './'

import { RkText } from 'react-native-ui-kitten'
import { Button } from 'react-native-elements'

export class AddToModal extends Component {
  constructor() {
    super()
    this.state = {
      isVisible: true,
    }
  }

  toggleModal () {
    this.state.isVisible = !this.state.isVisible
  }

  render () {
    const { card, list } = this.props
    return (
      <Modal isVisible onModalHide={() => this.toggleModal()} >
        <View style={styles.modal}>
          <RkText style={styles.title}>
            {`ADD ${card.name} TO WANT LIST`}
          </RkText>
          <Button
            large
            iconRight
            icon={{name: 'code'}}
            onPress={() => console.log('ADD CARD TO SOMETHING')}
            title='ADD'
          />
        </View>
      </Modal>
    )
  }
}

AddToModal.propTypes = {
  card: cardType.isRequired,
  listToAdd: PropTypes.any
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  title: {
    fontSize: 18,
    textAlign: 'center'
  },
  modal: {
    backgroundColor: 'white',
    height: 400
  }
})
