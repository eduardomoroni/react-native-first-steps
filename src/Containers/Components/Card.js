/* @flow */

import React, { Component } from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

type CardProps = {
  card: any
}

export default class Card extends Component {
  props: CardProps
  render () {
    const {
      name,
      text
    } = this.props.card
    return (
      <View style={styles.container}>
        <Text>{name}</Text>
        <Text>{text}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})
