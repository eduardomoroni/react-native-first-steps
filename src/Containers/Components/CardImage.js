/* @flow */

import React, { PureComponent } from 'react'
import {
  View,
  StyleSheet,
  Image,
  ActivityIndicator
} from 'react-native'

type CardImageProps = {
  card: any
}

type CardImageState = {
  loading: boolean
}

export class CardImage extends PureComponent {
  props: CardImageProps
  state: CardImageState

  constructor () {
    super()
    this.state = { loading: true }
  }

  render () {
    const { multiverseid } = this.props.card
    const IMG_URL = `http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=${multiverseid}&type=card`

    return (
      <View style={styles.container}>
        <ActivityIndicator size='large' style={this.state.loading ? styles.spinner : styles.hidden} />
        <Image
          style={this.state.loading ? styles.hidden : styles.image}
          source={{uri: IMG_URL}}
          onLoad={(e) => this.setState({loading: false})}
          onLoadStart={(e) => this.setState({loading: true})}
          onError={() => console.error('Error while loading card image')} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  image: {
    flex: 1
  },
  spinner: {
    flex: 1,
    justifyContent: 'center'
  },
  hidden: {
    width: 0,
    height: 0
  }
})
