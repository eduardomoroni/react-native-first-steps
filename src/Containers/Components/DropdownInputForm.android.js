/* @flow */

import React, { Component } from 'react'
import { View } from 'react-native'
import { InputLabel, InputPicker } from './'
import styles from '../../Styles/FormStyle'

export default class DropdownInputForm extends Component {
  render () {
    const { input, dropdownItems, selectedValue } = this.props
    const { onChange, name } = input

    return (
      <View style={styles.container}>
        <InputLabel label={name} />
        <InputPicker
          selectedValue={selectedValue}
          onValueChange={onChange}
          dropdownItems={dropdownItems}
        />
      </View>
    )
  }
}
