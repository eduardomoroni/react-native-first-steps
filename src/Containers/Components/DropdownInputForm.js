/* @flow */

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import {
  View,
  Text,
  Picker
} from 'react-native'
import styles from '../../Styles/FormStyle'

export default class DropdownInputForm extends Component {
  render () {
    const { input, dropdownItems, selectedValue } = this.props
    const { onChange, name } = input

    const renderPickerItem = (value, key) => {
      return <Picker.Item label={value} value={value} key={key} />
    }

    return (
      <View style={styles.container}>
        <Text style={styles.text}>
          {I18n.t(name)}
        </Text>
        <Picker
          style={styles.dropdown}
          selectedValue={selectedValue}
          onValueChange={onChange}>
          { dropdownItems.map(renderPickerItem) }
        </Picker>
      </View>
    )
  }
}
