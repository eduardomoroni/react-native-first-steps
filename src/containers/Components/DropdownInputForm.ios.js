/* @flow */

import React, { Component } from 'react'
import { View, TextInput } from 'react-native'
import { InputLabel } from './'
import styles from '../../styles/FormStyle'

// WIP
export class DropdownInputForm extends Component {
  render () {
    const { onChange, name } = this.props.input

    return (
      <View style={styles.container}>
        <InputLabel label={name} />
        <TextInput
          style={styles.input}
          onChangeText={onChange}
          autoCorrect={false}
          autoCapitalize='characters'
        />
      </View>
    )
  }
}

export default DropdownInputForm
