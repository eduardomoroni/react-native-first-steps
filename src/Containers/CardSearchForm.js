import React from 'react'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { connect } from 'react-redux'
import { realm } from '../Config/Realm'
import { inheritanceToArray } from '../Realm/Conversion/Realm-utils'
import { Metrics } from '../Styles/Themes'
import { searchForCards } from '../Redux/Actions'
import TextInputForm from './Components/TextInputForm'
import DropdownInputForm from './Components/DropdownInputForm'
import SubmitButtonForm from './Components/SubmitButtonForm'

import {
  StyleSheet,
  View,
  Keyboard
} from 'react-native'

type CardSearchFormProps = {
  cardTypes: any,
  cardSubTypes: any,
  cardType: string,
  cardSubType: string,
  searchCards: () => void,
  handleSubmit: any
}

const selector = formValueSelector('CardSearchForm')

let CardSearchForm = (props: CardSearchFormProps) => {
  const {
    handleSubmit,
    cardTypes,
    cardSubTypes,
    searchCards
  } = props

  const {
    cardType,
    cardSubType
  } = props

  const submit = values => {
    Keyboard.dismiss()
    searchCards(values)
  }

  return (
    <View style={Styles.container}>
      <Field name='cardName' component={TextInputForm} />
      <Field name='cardText' component={TextInputForm} />
      <Field name='cardType' component={DropdownInputForm} dropdownItems={cardTypes} selectedValue={cardType} />
      <Field name='cardSubType' component={DropdownInputForm} dropdownItems={cardSubTypes} selectedValue={cardSubType} />
      <SubmitButtonForm onPress={handleSubmit(submit)} />
    </View>
  )
}

Field.propTypes = {
  dropdownItems: React.PropTypes.arrayOf(React.PropTypes.object),
  selectedValue: React.PropTypes.string
}

const Styles = StyleSheet.create({
  container: {
    paddingTop: Metrics.navBarHeight + 15
  }
})

const mapStateToProps = (state) => {
  const cardTypes = inheritanceToArray(realm.objects('Type').snapshot())
  const cardSubtypes = inheritanceToArray(realm.objects('SubType').snapshot())

  cardTypes.unshift('')
  cardSubtypes.unshift('')

  return {
    cardTypes: cardTypes,
    cardSubTypes: cardSubtypes,
    cardName: selector(state, 'cardName'),
    cardType: selector(state, 'cardType'),
    cardSubType: selector(state, 'cardSubType'),
    cardText: selector(state, 'cardText')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    searchCards: (cardSearchForm) => dispatch(searchForCards(cardSearchForm))
  }
}

CardSearchForm = reduxForm({form: 'CardSearchForm'})(CardSearchForm)
export default connect(mapStateToProps, mapDispatchToProps)(CardSearchForm)
