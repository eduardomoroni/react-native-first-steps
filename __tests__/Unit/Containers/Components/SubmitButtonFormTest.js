// @flow

import 'react-native'
import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'
import { SubmitButtonForm } from '../../../../src/Containers/Components'

const props = {
  onPress: () => 'ABABAB'
}

it('Should render SubmitButtonForm Component', () => {
  const renderer = ReactTestUtils.createRenderer()
  const tree = renderer.render(<SubmitButtonForm {...props} />)
  expect(tree).toMatchSnapshot()
  expect(tree.props.onPress).toEqual(props.onPress)
})
