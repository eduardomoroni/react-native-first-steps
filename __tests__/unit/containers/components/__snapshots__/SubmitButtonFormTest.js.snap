// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButtonForm /> Snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3b5998",
      "height": 40,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontFamily": "beleren",
        "fontSize": 18,
      }
    }
  >
    I18n.t(search)
  </Text>
</TouchableOpacity>
`;
